pinc/misc.inc

```diff
@@ -298,41 +307,56 @@ function get_float_param($arr, $key, $default, $min, $max, $allownull = false)
  * @param int|float|null $max
  * @param bool $allownull
  *
+ * @return int|float|null
+ *
  * @throws InvalidArgumentException
  *
  * @see get_integer_param()
  * @see get_float_param()
  */
-function get_numeric_param($type, $arr, $key, $default, $min, $max, $allownull)
+function get_numeric_param(string $type, array $arr, string $key, $default, $min, $max, bool $allownull)
 {
-    {
-        // sanity checks on the args
-        assert(is_array($arr));
-        assert(!is_array($key));
-
-        if ($type == 'integer') {
-            assert(is_null($default) || is_int($default));
-        } else {
-            assert(is_null($default) || is_float($default));
+    // sanity checks on the args
+    if ($type == 'integer') {
+        if (! (is_null($default) || is_int($default))) {
+            throw new TypeError("\$default must be of the type int or null");
```

Comment:
If you wanted to make this marginally more efficient, you could pull the is_null($default) to the top level and say...
```
if (!is_null($default)) {
   if ($type == 'integer') {
     if (!is_int($default)) { throw ... }
   } elseif ($type == 'float') {
    if  (!is_float($default)) { throw ... }
   } else {
    throw "invalid type..."
   }
}
```

Reply:
That would only validate `$type` in the case that `$default` is not null though.

Reply:
Oh. Yeah, true!